src/rpp/rpp/operators/fwd/switch_map.hpp

```diff
@@ -0,0 +1,67 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2022 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/observables/constraints.hpp>
+#include <rpp/observables/details/member_overload.hpp>
+
+namespace rpp::details
+{
+struct switch_map_tag;
+}
+
+namespace rpp::details
+{
+template<typename Fn, typename Type>
+concept switch_map_callable = std::invocable<Fn, Type> && constraint::observable<std::invoke_result_t<Fn, Type>>;
+
+template<constraint::decayed_type Type, switch_map_callable<Type> Callable>
+auto switch_map_impl(auto&& observable, Callable&& callable);
+
+template<constraint::decayed_type Type, typename SpecificObservable>
+struct member_overload<Type, SpecificObservable, switch_map_tag>
+{
+    /**
+    * \brief Transform emissions to observables via provided function and then merge emissions from such an observables
```

Comment:
Please, change description to actual one (you can take some part from `switch_on_next`)